src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/SharedArrayBufferTests.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices.JavaScript;
+using Xunit;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    public static class SharedArrayBufferTests
+    {
+        private static Function _objectPrototype;
+
+        public static IEnumerable<object[]> Object_Prototype()
+        {
+            _objectPrototype ??= new Function("return Object.prototype.toString;");
+            yield return new object[] { _objectPrototype.Call() };
+        }
+
+        [Theory]
+        [MemberData(nameof(Object_Prototype))]
+        public static void SharedArrayBufferLength(Function objectPrototype)
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(0);
+            Assert.Equal("[object SharedArrayBuffer]", objectPrototype.Call(d));
+            Assert.Equal(0, d.Length);
+        }
+
+        [Theory]
+        [MemberData(nameof(Object_Prototype))]
+        public static void SharedArrayBufferLength1(Function objectPrototype)
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(50);
+            Assert.Equal("[object SharedArrayBuffer]", objectPrototype.Call(d));
+            Assert.Equal(0, d.Length);
+        }
+
+        [Fact]
+        public static void SharedArrayBufferLength2()
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(50);
+            Assert.Equal(50, d.ByteLength);
+        }
+
+        [Fact]
+        public static void SharedArrayBufferSlice()
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(50);
+            Assert.Equal(50, d.Slice().ByteLength);
+        }
+
+        [Fact]
+        public static void SharedArrayBufferSlice1()
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(50);
+            Assert.Equal(50, d.Slice(0, 50).ByteLength);
+        }
+
+        [Fact]
+        public static void SharedArrayBufferSlice2()
+        {
+            SharedArrayBuffer d = new SharedArrayBuffer(50);
+            Assert.Equal(50, d.Slice(0).ByteLength);
+        }
+
+        [Fact]
+        public static void SharedArrayBufferSlice3()
```

Comment:
Maybe SharedArrayBuffer_Slice_BeginNegative?

Reply:
Addressed in recent commits